pyControl4/room.py

```diff
@@ -0,0 +1,140 @@
+"""Controls Control4 Room devices.
+"""
+
+
+from pyControl4 import C4Entity
+
+
+class C4Room(C4Entity):
+    """
+    A media-oriented view of a Control4 Room, supporting items of type="room"
+    """
+
+    async def isRoomHidden(self) -> bool:
+        """Returns True if the room is hidden from the end-user"""
+        value = await self.director.getItemVariableValue(self.item_id, "ROOM_HIDDEN")
+        return int(value) != 0
+
+    async def isOn(self) -> bool:
+        """Returns True/False if the room is "ON" from the director's perspective"""
+        value = await self.director.getItemVariableValue(self.item_id, "POWER_STATE")
+        return int(value) != 0
+
+    async def setRoomOff(self):
+        """Turn the room "OFF" """
+        await self.director.sendPostRequest(
+            "/api/v1/items/{}/commands".format(self.item_id),
+            "ROOM_OFF",
+            {},
+        )
+
+    async def setSource(self, source_id: int, audio_only: bool):
+        """
+        Sets the room source, turning on the room if necessary.
+        If audio_only, only the current audio device is changed
+        """
+        await self.director.sendPostRequest(
+            f"/api/v1/items/{self.item_id}/commands",
+            "SELECT_AUDIO_DEVICE" if audio_only else "SELECT_VIDEO_DEVICE",
+            {"deviceid": source_id},
+        )
+
+    async def setAudioSource(self, source_id: int):
+        """Sets the current audio source for the room"""
+        await self.setSource(source_id, audio_only=True)
+
+    async def setVideoAndAudioSource(self, source_id: int):
+        """Sets the current audio and video source for the room"""
+        await self.setSource(source_id, audio_only=False)
```

Comment:
Since we have two different set audio/set audio and video methods, the `setSource` method should be private (`_setSource`)?